kurtosis-devnet/tests/isthmus/erc20_bridge_test.go

```diff
@@ -0,0 +1,214 @@
+		require.NoError(t, err)
+
+		// Create the L2 token
+		l2TokenName := "L2 Test Token"
+		l2TokenSymbol := "L2TEST"
+		tx, err = optimismMintableTokenFactory.CreateOptimismMintableERC20(l2Opts, l1TokenAddress, l2TokenName, l2TokenSymbol)
+		require.NoError(t, err)
+		l2TokenReceipt, err := wait.ForReceiptOK(ctx, l2Client, tx.Hash())
+		require.NoError(t, err, "Failed to create L2 token")
+
+		// Extract the L2 token address from the event logs
+		var l2TokenAddress common.Address
+		for _, log := range l2TokenReceipt.Logs {
+			createdEvent, err := optimismMintableTokenFactory.ParseOptimismMintableERC20Created(*log)
+			if err == nil && createdEvent != nil {
+				l2TokenAddress = createdEvent.LocalToken
+				break
+			}
+		}
+		require.NotEqual(t, common.Address{}, l2TokenAddress, "Failed to find L2 token address from events")
+		logger.Info("Created L2 token", "address", l2TokenAddress)
+
+		// Get the L2 token contract
+		l2Token, err := bindings.NewOptimismMintableERC20(l2TokenAddress, l2Client)
+		require.NoError(t, err)
+
+		// Check initial L2 token balance (should be 0)
+		initialL2Balance, err := l2Token.BalanceOf(&bind.CallOpts{}, l2User.Address())
+		require.NoError(t, err)
+		require.True(t, big.NewInt(0).Cmp(initialL2Balance) == 0, "Initial L2 token balance should be 0, actual was %s", initialL2Balance.String())
+
+		l1StandardBridgeAddress, ok := l2Chain.Addresses()["l1StandardBridgeProxy"]
+		require.True(t, ok, fmt.Errorf("no L1 proxy address configured for this test"))
+
+		l1StandardBridge, err := bindings.NewL1StandardBridge(l1StandardBridgeAddress, l1Client)
+		require.NoError(t, err)
+
+		// Approve the L1 bridge to spend tokens
+		logger.Info("Approving L1 bridge to spend tokens")
+		tx, err = l1Token.Approve(l1Opts, l1StandardBridgeAddress, mintAmount)
+		require.NoError(t, err)
+		_, err = wait.ForReceiptOK(ctx, l1Client, tx.Hash())
+		require.NoError(t, err, "Failed to approve L1 bridge")
+
+		// Amount to bridge
+		bridgeAmount := big.NewInt(100000000000000000) // 0.1 token
+		minGasLimit := uint32(200000)                  // Minimum gas limit for the L2 transaction
+
+		// Bridge the tokens from L1 to L2
+		logger.Info("Bridging tokens from L1 to L2", "amount", bridgeAmount)
+		tx, err = l1StandardBridge.DepositERC20To(
+			l1Opts,
+			l1TokenAddress,
+			l2TokenAddress,
+			l2User.Address(),
+			bridgeAmount,
+			minGasLimit,
+			[]byte{}, // No extra data
+		)
+		require.NoError(t, err)
+		depositReceipt, err := wait.ForReceiptOK(ctx, l1Client, tx.Hash())
+		require.NoError(t, err, "Failed to deposit tokens to L2")
+		logger.Info("Deposit transaction confirmed on L1", "tx", tx.Hash().Hex())
+
+		// Get the OptimismPortal contract to find the deposit event
+		optimismPortalAddress := common.HexToAddress("0x4200000000000000000000000000000000000000") // Standard address for OptimismPortal
```

Comment:
`0x4200000000000000000000000000000000000000` is `LegacyMessagePasser`, which can be imported from devnet-sdk/contracts/constants(we can add the constants there). But curious about this actual value because OptimismPortal is not a predeploy but a contract living in L1.

Reply:
So when we trigger a deposit

Reply:
Good catch, not sure how this got here.